### PR TITLE
Rename DOM types and change namespace

### PR DESCRIPTION
**Pull request by rynowak**
Fixes: #12553

This change renames all of our browser/DOM specific types from
`UIFooEventArgs` to `FooEventArgs` and puts the in the `.Web` namespace.

In addition to this, we're moving `EventHandlers` and `BindAttributes`
to the same. This has the impact of scoping the mappings those classes
provide based on the `.Web` namespace.

This means that we now expect `.Web` to be present as a using in
basically all contexts for a browser-based Blazor app. Updated
templates, samples and tests. I'll also need to update about a million
tests in the compiler codebase.

I've logged https://github.com/aspnet/AspNetCore.Docs/issues/13832 to
track the docs and release notes part of this work.

### CODE REVIEW

File: src/Components/Web/src/Web/EventHandlers.cs [hunk 1]
```diff
@@ -0,0 +1,128 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.AspNetCore.Components.Web
```
Comment (rynowak) [inline src/Components/Web/src/Web/EventHandlers.cs]: It turns out the compiler already treats event handlers and bind attributes as scoped based on namespace. I wasn't aware this already worked, and I figured we'd have to add that support later.

This is important because it allows us control how a feature like `bind` behaves based on the usings. Suppose we had *blazor native* for some other UI stack, it won't offer completion for DOM based events because they won't be in scope, because you won't have a using for `.Web`.

It's important that this behaviour is controller by usings rather than assembly references, because user code or libraries end up with extra references all of the time by accident.

File: src/Components/Blazor/Build/test/BindRazorIntegrationTest.cs [hunk 2]
```diff
@@ -271,6 +271,7 @@ public void Render_BuiltIn_BindToInputWithoutType_WritesAttributes()
         {
             // Arrange
             var component = CompileToComponent(@"
+@using Microsoft.AspNetCore.Components.Web
```
Comment (pranavkm) [inline src/Components/Blazor/Build/test/BindRazorIntegrationTest.cs]: Do we need to add this namespace to https://github.com/aspnet/AspNetCore/blob/release/3.0/src/ProjectTemplates/Web.ProjectTemplates/content/BlazorServerWeb-CSharp/_Imports.razor#L5?

Comment (pranavkm) [reply src/Components/Blazor/Build/test/BindRazorIntegrationTest.cs]: Also client side templates?

**Review by pranavkm — APPROVED** [review]
One question:

File: src/Components/Blazor/Build/test/BindRazorIntegrationTest.cs (hunk 2 above)
Comment (rynowak) [reply src/Components/Blazor/Build/test/BindRazorIntegrationTest.cs]: That's wierd, I thought I got them both. Will take a second look 😆

Comment (rynowak) [reply src/Components/Blazor/Build/test/BindRazorIntegrationTest.cs]: Am I crazy? https://github.com/aspnet/AspNetCore/pull/13138/files#diff-ab56cd4be78548275aa24134d6d8b06aR6

Comment (pranavkm) [reply src/Components/Blazor/Build/test/BindRazorIntegrationTest.cs]: Guess not. Weird I thought that missing when I first saw it. This looks ok

Comment (rynowak) [reply src/Components/Blazor/Build/test/BindRazorIntegrationTest.cs]: I added it to the classlib template where it was missing.

**Review by NTaylorMullen — APPROVED** [review]

**Comment by pranavkm** [comment]
@aspnet-hello triage https://dev.azure.com/dnceng/public/_build/results?buildId=310639&view=logs&jobId=a89a2a8c-213d-51e6-fe71-933481fa439c

**Comment by aspnet-hello** [comment]
This comment was made automatically. If there is a problem contact aspnetcore-build@microsoft.com.

I've triaged the above build. I've created/commented on the following issue(s)
https://github.com/aspnet/AspNetCore-Internal/issues/2982

**Comment by pranavkm** [comment]
@rynowak you broke some template tests

**Comment by rynowak** [comment]
Totes mah gotes. Haven't had time to fix it yet 😆
